EpiAware/src/EpiInfModels/ODEProcess.jl

```diff
@@ -0,0 +1,253 @@
+
+# Define the ODE problem using SciML
+prob = ODEProblem(expgrowth, [1.0], (0.0, 10.0), [r])
+
+# Define the custom parameters struct
+struct CustomParams <: AbstractTuringParamModel
+    prob::ODEProblem
+    r::Float64
+    u0::Float64
+end
+params = CustomParams(prob, r, 1.0)
+
+# Define the custom generate_latent function
+@model function EpiAwareBase.generate_latent(params::CustomParams, Z_t)
+    return ([params.u0], [params.r])
+end
+```
+
+This model is not random! But we can still use it to generate latent infections.
+
+```julia
+# Define the ODEProcess
+expgrowth_model = ODEProcess(
+    params = params,
+    sol2infs = sol -> sol[1, :]
+)
+infs = generate_latent_infs(expgrowth_model, nothing)()
+```
+"""
+@kwdef struct ODEProcess{
+    P <: AbstractTuringLatentModel, S, F <: Function, D <:
+                                                      Union{Dict, NamedTuple}} <:
+              EpiAwareBase.AbstractTuringEpiModel
+    "The ODE problem instance, where `P` is a subtype of `ODEProblem`."
+    params::P
+    "The solver used for the ODE problem. Default is `AutoVern7(Rodas5())`, which is an auto
+    switching solver aimed at medium/low tolerances."
+    solver::S = AutoVern7(Rodas5())
+    "A function that maps the solution object of the ODE to infection counts."
+    sol2infs::F
+    "The extra solver options for the ODE problem. Can be either a `Dict` or a `NamedTuple`
+    containing the solver options."
+    solver_options::D = Dict(:verbose => false, :saveat => 1.0)
+end
+
+@doc raw"""
+Implement the `generate_latent_infs` function for the `ODEProcess` model.
+
+This function remakes the ODE problem with the provided initial conditions and parameters,
+    solves it using the specified solver, and then transforms the solution into latent infections
+    using the `sol2infs` function.
+
+# Example usage with predefined SIR model
+
+In this example we define an `ODEProcess` object using the predefined `SIRParams` model and
+generate an expected infection time series using SIR model parameters sampled from their priors.
+
+```julia
+using EpiAware, OrdinaryDiffEq, Distributions, LogExpFunctions
+
+# Create an instance of SIRParams
+sirparams = SIRParams(
+    tspan = (0.0, 30.0),
+    infectiousness = LogNormal(log(0.3), 0.05),
+    recovery_rate = LogNormal(log(0.1), 0.05),
+    initial_prop_infected = Beta(1, 99)
+)
+#Population size
+N = 1000.0
+sir_process = ODEProcess(
+    params = sirparams,
+    sol2infs = sol -> N .* logaddexp.(0.001, sol[2, :]),
+    solver_options = Dict(:verbose => false, :saveat => 1.0)
+)
+
+generated_It = generate_latent_infs(sir_process, nothing)()
+```
+
+"""
+@model function EpiAwareBase.generate_latent_infs(epi_model::ODEProcess, Z_t)
+    prob, solver, sol2infs, solver_options = epi_model.params.prob,
+    epi_model.solver, epi_model.sol2infs, epi_model.solver_options
+
+    @submodel prefix="params" u0, p=generate_latent(epi_model.params, Z_t)
```

Comment:
why apply a prefix here?

Reply:
Left over from the `generate_parameter` approach

Reply:
And gone now.